src/codegen/llvm/codegen_llvm_helper_visitor.cpp

```diff
@@ -474,16 +475,56 @@ void CodegenLLVMHelperVisitor::convert_local_statement(ast::StatementBlock& node
 
         /// remove local list statement now
         std::unordered_set<nmodl::ast::Statement*> to_delete({local_statement.get()});
-        node.erase_statement(to_delete);
+        /// local list statement is enclosed in statement block
+        const auto& parent_node = dynamic_cast<ast::StatementBlock*>(local_statement->get_parent());
+        parent_node->erase_statement(to_delete);
 
         /// create new codegen variable statement and insert at the beginning of the block
         auto type = new ast::CodegenVarType(FLOAT_TYPE);
-        auto statement = std::make_shared<ast::CodegenVarListStatement>(type, variables);
-        const auto& statements = node.get_statements();
-        node.insert_statement(statements.begin(), statement);
+        auto new_statement = std::make_shared<ast::CodegenVarListStatement>(type, variables);
+        const auto& statements = parent_node->get_statements();
+        parent_node->insert_statement(statements.begin(), new_statement);
+    }
+}
+
+/**
+ * \brief Visit StatementBlock and rename all LOCAL variables
+ * @param node AST node representing Statement block
+ *
+ * Statement block in remainder loop will have same LOCAL variables from
+ * main loop. In order to avoid conflict during lookup, rename each local
+ * variable by appending unique number. The number used as suffix is just
+ * a counter used for Statement block.
+ *
+ * \todo : Check in LLVM if it is really necessary to rename all local
+ *         variables across basic blocks. There should be a concept
+ *         of local symbol table so that we don't need to have unique
+ *         variable names?
+ */
+void CodegenLLVMHelperVisitor::rename_local_variables(ast::StatementBlock& node) {
+    /// local block counter just to append unique number
+    static int local_block_counter = 1;
+
+    /// collect all local statement block
+    const auto& statements = collect_nodes(node, {ast::AstNodeType::LOCAL_LIST_STATEMENT});
+
+    /// iterate over each statement and rename all variabbles
+    for (const auto& statement: statements) {
+        const auto& local_statement = std::dynamic_pointer_cast<ast::LocalListStatement>(statement);
+
+        /// rename local variable in entire statement block
+        for (auto& var: local_statement->get_variables()) {
```

Comment:
I was thinking to extend `RenameVisitor` to rename the vector of variables or names with a prefix, I think there are other places where we have to iterate and call the visitor each time. But that's for separate PR and is not necessary at all yet (and not a priority).

Reply:
👍 - would keep this in a separate PR